schema.ts

```diff
@@ -1321,9 +1348,62 @@ function checkSchemaPropertyIsRequired<K extends string, L extends string, T ext
   );
 }
 
+/**
+ * Normalizes a schema property key into PascalCase. This interprets "."s as accessing object properties
+ * and "[]" as accessing array items.
+ */
 export function normalizeSchemaKey(key: string): string {
-  // Colons cause problems in our formula handling.
-  return pascalcase(key).replace(/:/g, '_');
+  // Try splitting by . to handle json paths.
+  return (
+    key
+      .split('.')
+      // Colons cause problems in our formula handling.
+      .map(val => {
+        let partToNormalize = val;
+        let partToIgnoreNormalization = '';
+
+        // Handles array pathing.
+        if (val.includes('[')) {
+          partToNormalize = val.substring(0, val.indexOf('['));
+          partToIgnoreNormalization = val.substring(val.indexOf('['));
+        }
+
+        return (pascalcase(partToNormalize) + partToIgnoreNormalization).replace(/:/g, '_');
+      })
+      .join('.')
+  );
+}
+
+/**
+ * Normalizes a schema PropertyIdentifier by converting it to PascalCase.
+ */
+function normalizeSchemaPropertyIdentifier(key: PropertyIdentifier): PropertyIdentifier {
+  if (typeof key === 'string') {
+    return normalizeSchemaKey(key);
+  }
+
+  const {label, property: value} = key;
+  return {
+    property: normalizeSchemaKey(value),
+    label: normalizeSchemaKey(label),
```

Comment:
Should this just get regular pascal casing? This isn't a path so it probably shouldn't be subject to the carveout for dots and braces right?